Networking/CommunicationManager/ServerCommunicator.cs

```diff
@@ -0,0 +1,160 @@
+using System;
+using System.Threading;
+using System.Net.Sockets;
+using System.Text;
+using System.Collections;
+using System.Collections.Generic;
+using System.Net;
+using System.Diagnostics;
+
+/// <summary>
+/// This file contains the class Server which will be running on the server.
+/// and accepts client request.
+/// </summary>
+/// <author>Tausif Iqbal </author>
+namespace Networking
+{
+    public class ServerCommunicator : ICommunicator
+    {
+        private Dictionary<string, INotificationHandler> _subscribedModules = new Dictionary<string, INotificationHandler>();
+        private Hashtable _clientIdSocket = new Hashtable();
+        private Queue _recieveQueue = new Queue();
+
+        private Thread _acceptRequest;
+        private bool _acceptRequestRun = false;
+        private SendSocketListenerServer _sendSocketListenerServer;
+       
+       private HashSet<RecieveSocketListener> clientListener = new HashSet<RecieveSocketListener>();
+
+        /// <summary>
+        /// It finds IP4 address of machine which does not end with .1
+        /// </summary>
+        /// <returns>IP4 address </returns>
+        private static string GetLocalIPAddress()
+        {
+            var host = Dns.GetHostEntry(Dns.GetHostName());
+            foreach (var ip in host.AddressList)
+            {
+                if (ip.AddressFamily == AddressFamily.InterNetwork)
+                {
+                    string address = ip.ToString();
+                    int n = address.Length;
+                    if( address[n-1]!='1' && address[n - 2] != '.')
```

Comment:
The condition is wrong, verify and make PR again.

Reply:
Changed the condition by splitting the string and checking the last part alone.